alpa/pipeline_parallel/pipeshard_executable.py

```diff
@@ -90,12 +122,83 @@ def __init__(self,
         self.batch_invars = input_config.batch_invars
 
         ##### For handling outputs of the executable #####
+        #TODO Github Task - commenting these lines
         self.output_local_uuid_list = pipeshard_config.output_local_uuid_list
         self.outs_handler = pipeshard_config.outs_handler
 
+
+        #TODO Github task -adding this line
+        virtual_to_pysical_map = {}
+        temp_worker_to_rank_map = {}
+        #if pipeshard_config.vworker:
+        self.mesh_group.collective_groups = pipeshard_config.collective_grp
+        #TODO Github task - replacing virtual workers with ray workers
+        temp_mesh_grp = []
+        for mesh in self.mesh_group.meshes:
+            for worker in mesh.workers:
+                temp_mesh_grp.append(worker)
+        temp_worker_to_rank_map = {
+            worker: r for r, worker in enumerate(temp_mesh_grp)
+        }
+        for cgp in self.mesh_group.collective_groups:
+            for cg in cgp:
+                if cg is not None:
+                    cg.mesh_workers = temp_mesh_grp
+                    cg.worker_to_rank_map = temp_worker_to_rank_map
+        for virtual_worker, _ in pipeshard_config.instruction_lists.items():
+            virtual_to_pysical_map[virtual_worker.index] = virtual_worker
```

Comment:
Move codes to launch on real physical resources into a method of `VirtualMeshGroup`(probably name it `instantiate`). The target of the function is to lower a `VirtualMeshGroup` into a `PhysicalDeviceMeshGroup`. The pseudo-code of the method is:
```python
def instantiate(self, *args):
    if exist_a_physical_mesh_group():
        existing_group = get_existing_physical_mesh_group()
        self._check_compatibility(existing_group)
        return existing group
    else:
        return self._instantiate_mesh_group()
```